packages/forms/src/fields/MultiThumbRange.js

```diff
@@ -0,0 +1,428 @@
+        diffX *= -1;
+      }
+
+      let newValue = min + parseInt(((max - min) * diffX) / trackWidth, 10);
+
+      // Reduce updated value to align with step size
+      newValue = Math.floor(newValue / step) * step;
+
+      if (isMinThumbFocused) {
+        updateMinThumbSlider(newValue);
+      } else {
+        updateMaxThumbSlider(newValue);
+      }
+    },
+    [isMinThumbFocused, max, min, rtl, step, updateMaxThumbSlider, updateMinThumbSlider]
+  );
+
+  const removeDragEvents = useCallback(() => {
+    themedDocument.removeEventListener('mousemove', onDocumentMouseMove);
+    themedDocument.removeEventListener('mouseup', removeDragEvents);
+
+    setIsMousedDown(false);
+  }, [onDocumentMouseMove, themedDocument]);
+
+  useEffect(() => {
+    if (isMousedDown) {
+      themedDocument.addEventListener('mousemove', onDocumentMouseMove);
+      themedDocument.addEventListener('mouseup', removeDragEvents);
+    }
+
+    return () => {
+      if (isMousedDown) {
+        themedDocument.removeEventListener('mousemove', onDocumentMouseMove);
+        themedDocument.removeEventListener('mouseup', removeDragEvents);
+      }
+    };
+  }, [isMousedDown, onDocumentMouseMove, removeDragEvents, themedDocument]);
+
+  const onKeyDown = type => e => {
+    const isMinThumb = type === 'min';
+    let keyIntercepted = false;
+
+    const decrementThumb = () => {
+      if (isMinThumb) {
+        updateMinThumbSlider(minValue - step);
+      } else {
+        updateMaxThumbSlider(maxValue - step);
+      }
+    };
+
+    const incrementThumb = () => {
+      if (isMinThumb) {
+        updateMinThumbSlider(minValue + step);
+      } else {
+        updateMaxThumbSlider(maxValue + step);
+      }
+    };
+
+    switch (e.keyCode) {
+      case KEY_CODES.LEFT:
+        if (rtl) {
+          incrementThumb();
+        } else {
+          decrementThumb();
+        }
+
+        keyIntercepted = true;
+        break;
+
+      case KEY_CODES.DOWN:
+        decrementThumb();
+        keyIntercepted = true;
+        break;
+
+      case KEY_CODES.RIGHT:
+        if (rtl) {
+          decrementThumb();
+        } else {
+          incrementThumb();
+        }
+
+        keyIntercepted = true;
+        break;
+
+      case KEY_CODES.UP:
+        incrementThumb();
+        keyIntercepted = true;
+        break;
+
+      case KEY_CODES.HOME:
+        if (isMinThumb) {
+          updateMinThumbSlider(min);
+        } else {
+          updateMaxThumbSlider(min);
+        }
+
+        keyIntercepted = true;
+        break;
+
+      case KEY_CODES.END:
+        if (isMinThumb) {
+          updateMinThumbSlider(max);
+        } else {
+          updateMaxThumbSlider(max);
+        }
+
+        keyIntercepted = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (keyIntercepted) {
+      e.preventDefault();
+      e.stopPropagation();
+    }
+  };
+
+  // This may be increased to enforce separation between thumbs
+  const MIN_DISTANCE_PX = 0;
```

Comment:
This only affects the visual distance between elements, not the actual values returned. i.e. this could be used to only allow the edges of the thumbs to touch.

When we went over the original design for this we decided to allow overlapping since both values could technically be the same.